pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2337,3 +2337,34 @@ def test_merge_join_cols_error_reporting_on_and_index(func, kwargs):
     )
     with pytest.raises(MergeError, match=msg):
         getattr(pd, func)(left, right, on="a", **kwargs)
+
+
+@pytest.mark.parametrize(
+    ("input_col", "output_cols"), [("b", ["a", "b"]), ("a", ["a_x", "a_y"])]
+)
+def test_merge_cross(input_col, output_cols):
+    # GH#5401
+    left = DataFrame({"a": [1, 3]})
+    right = DataFrame({input_col: [3, 4]})
+    result = merge(left, right, how="cross")
+    expected = DataFrame({output_cols[0]: [1, 1, 3, 3], output_cols[1]: [3, 4, 3, 4]})
+    tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "kwargs",
```

Comment:
can you als assert that the original frames are unchanged (in fact do an 'is' assert as they shouldn't be touched at all)

Reply:
Sorry, but could you elaborate further, what you are reffering to with ``is`` assert?

Reply:
```
left_copy = left.copy()
....
# do cross

# check we did not mutate inputs
tm.assert_frame_equal(left, left_copy)
```
and same for right

Reply:
you can do this in test_merge_cross

Reply:
Done